sdk/boms/azure-sdk-bom/pom.xml

```diff
@@ -47,322 +44,259 @@
       <dependency>
         <groupId>com.azure</groupId>
         <artifactId>azure-ai-formrecognizer</artifactId>
-        <version>3.0.8</version>
+        <version>3.1.1</version>
       </dependency>
-
       <dependency>
         <groupId>com.azure</groupId>
         <artifactId>azure-ai-textanalytics</artifactId>
-        <version>5.0.6</version>
+        <version>5.0.7</version>
       </dependency>
-
-      <dependency>
-        <groupId>com.azure</groupId>
-        <artifactId>azure-communication-chat</artifactId>
```

Comment:
Depends on azure-core 1.14

Reply:
I think the tooling should take into account libraries that were included in previous BOM releases, removal of this could result in a compilation error when upgrading

Reply:
Yes and it is one of the goals. However - I do not want the tool to be doing that.-This comes from the report instead and becomes the start of the conversation with the partner team to have them upgrade.

In case, the library is not upgraded - I would think it will be best to not put it in the BOM. Just because even though the customer would get errors - it is still more explicit than using the library and then seeing cryptic failures at compile or runtime. Beats the purpose of the BOM all together.

Reply:
Should the tool even generate a BOM when this happens? We should just generate a report and include all libraries that were dropped. Generating a partial BOM is not required as it is not valid and can only increase chances of accidentally releasing it.

Reply:
The tool does not update the BOM in place (at the moment at least). It always generates it as a separate file.

> Should the tool even generate a BOM when this happens

This is a tough one. One one side - it should not, since we should not have breaking changes in BOM. On the other hand, do we not release a BOM at all - if one of the dependency is not updated on time?

Reply:
Yes, we should not release the BOM at all if one of the dependencies gets dropped. It is a ship blocker if we remove any library that existed in a previous release.

Reply:
Do we follow the same guidance for any external library that is in the BOM? Say for example - netty-tcnative, oauth2-oidc-sdk?

Reply:
Thought about this a little bit and here is what I would like us start with and change gears if needed later on.

Internal libraries - all azure SDKs - we never drop them. We try to converge as much as we can but bring along the ones that don't converge.
External libraries like histogram, oauth2 etc - we drop and bring in as needed. With enough time, I am hoping we will learn which to not put in the BOM at all.
core-test - I will check if I there is telemetry data to get usage but ultimately would like to remove it from the BOM. I will work with spring team and decouple the dependency. Having external customers take accidental dependency on core-test  is not a great story.

Reply:
Added back the dropped SDK dependencies.

Reply:
> External libraries like histogram, oauth2 etc - we drop and bring in as needed. With enough time, I am hoping we will learn which to not put in the BOM at all.

Wouldn't dropping external libraries also be a breaking change? However, if all Azure SDK libraries are aligned, I think in most cases external libraries will also be aligned because the external_libraries.txt will generally include just one version.

Reply:
Yes but there is a possibility that we may have stale external binaries in our BOM which we may no longer need. Also, I feel like there is going to be a learning curve here - should all external libraries be in the BOM? What if it is used by one library for something super niche and do we include all the 10 niche libraries that 10 GA libraries use? Keeping this fungible gives us flexibility.